server/game/cards/07-WotW/ShinomenWayfinders.js

```diff
@@ -0,0 +1,25 @@
+const DrawCard = require('../../drawcard.js');
+const { Locations, Players, PlayTypes } = require('../../Constants');
+const AbilityDsl = require('../../abilitydsl.js');
+
+class ShinomenWayfinders extends DrawCard {
+    setupCardAbilities() {
+        this.persistentEffect({
+            location: Locations.PlayArea,
+            targetController: Players.Self,
+            effect: AbilityDsl.effects.reduceCost({
+                playingTypes: PlayTypes.PlayFromHand,
+                amount: context => {
+                    return context.player.filterCardsInPlay(card => {
+                        return card.isParticipating() && card.isFaction('unicorn');
+                    }).length;
+                },
```

Comment:
A couple of problems here:
- I think the cost reduction should work from anywhere
- It should function for both players (if I use infiltrator to play your Wayfinders, I get the reduction too)
- I think it should effect both PlayTypes (I should get the reduction if I play it from BAH - I guess that doesn't really matter, but you never know what they will print in the future)
- I think amount can be a function, but it takes the card being played as it's parameter, not the context object.  That might be problematic in the case where it's played using Infiltrator...

Reply:
so I added the things you asked for expect I made the context the only parameter to the function for the amount. I am not really sure how you would determine the number of attacking characters without having the context